Homework_1_SergioFG

```diff
@@ -0,0 +1,116 @@
+##################################################################################################
+
+# check my data 
+data_iris.head
+data_iris.columns = ["Sepal_Length", "sepal_width", "petal_length", "petal_width", "Iris class"]
+fig = px.scatter_matrix(data_iris, dimensions=["sepal_width", "Sepal_Length"
+                                    , "petal_width", "petal_length"], color="Iris class")
+fig.show()
+
+
+##################################################################################################
+#                                          Density Contour                                       #
+##################################################################################################
+
+data_iris.columns = ["Sepal_Length", "sepal_width", "petal_length", "petal_width", "species"]
+fig = px.density_contour(data_iris, x="sepal_width", y="Sepal_Length")
+fig.show()
+
+##################################################################################################
+#                                      Scikit Learn a go-go                                      #
+##################################################################################################
+
+from sklearn.preprocessing import Normalizer
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.datasets import make_classification
+# Normalizer doesn't work with strings so kept it to numerical data
+X = data_iris [["Sepal_Length", "sepal_width", "petal_length", "petal_width"]]
+transformer = Normalizer().fit(X)
+#set norm to mean of sepal length and transform it by that
+#transformer = Normalizer(norm=X["Sepal_Length"].mean).transform(X)
+transformer
+X_trans = transformer.transform(X)
+X_trans
+#tried to adjust this to # of rows and columns
+#X_trans, y = make_classification(n_samples=149, n_features=4, n_informative=2, n_redundant=0, random_state=0, shuffle=False)
+X_trans, y = make_classification(n_samples=149, n_features=4)
+clf = RandomForestClassifier(max_depth=4, random_state=0)
+clf.fit(X_trans,y)
+print(clf.predict([[1,0,0,0]]))
+
+
+##################################################################################################
+#                                      Pipelining                                                #
+##################################################################################################
+
+from sklearn.svm import SVC
+from sklearn.model_selection import train_test_split
+from sklearn.pipeline import Pipeline
+X_trans, y = make_classification(random_state=0)
+X_train, X_test, y_train, y_test = train_test_split(X_trans, y, random_state=0)
+pipe = Pipeline([('scaler', Normalizer()), ('svc', SVC())])
+
+# The pipeline can be used as any other estimator
+# and avoids leaking the test set into the train set
+### I split the data to avoid data leackage. Not that it matters but it runs :p 
+
+pipe.fit(X_train, y_train)
+ 
+pipe.score(X_test, y_test)
```

Comment:
Might be helpful to save your end results.  Otherwise looks good

Reply:
Will do so.